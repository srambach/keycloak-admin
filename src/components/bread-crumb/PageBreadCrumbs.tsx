import React from "react";
import { Link } from "react-router-dom";
import useBreadcrumbs from "use-react-router-breadcrumbs";
import { useTranslation } from "react-i18next";
import { Breadcrumb, BreadcrumbItem } from "@patternfly/react-core";

import { routes } from "../../route-config";

export const PageBreadCrumbs = () => {
  const { t } = useTranslation();
  const crumbs = useBreadcrumbs(routes(t)).slice(1);
  return (
    <>
      {crumbs.length > 1 && (
        <Breadcrumb>
          {crumbs.map(({ match, breadcrumb: crumb }, i) => (
            <BreadcrumbItem key={i} isActive={crumbs.length - 1 === i}>
              {crumbs.length - 1 !== i && <Link to={match.url}>{crumb}</Link>}
              {crumbs.length - 1 === i && <>{crumb}</>}
            </BreadcrumbItem>
          ))}
        </Breadcrumb>
      )}
    </>
  );
};
