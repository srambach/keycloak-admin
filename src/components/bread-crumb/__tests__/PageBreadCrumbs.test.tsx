import React from "react";
import { mount } from "enzyme";
import { PageBreadCrumbs } from "../PageBreadCrumbs";
import { MemoryRouter } from "react-router-dom";

describe("BreadCrumbs tests", () => {
  it("couple of crumbs", () => {
    const crumbs = mount(
      <MemoryRouter initialEntries={["/clients/1234"]}>
        <PageBreadCrumbs />
      </MemoryRouter>
    );
    expect(crumbs.find(PageBreadCrumbs)).toMatchSnapshot();
  });
});
