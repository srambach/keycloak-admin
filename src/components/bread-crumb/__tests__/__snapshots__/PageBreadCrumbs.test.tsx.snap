// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BreadCrumbs tests couple of crumbs 1`] = `
<PageBreadCrumbs>
  <Breadcrumb>
    <nav
      aria-label="Breadcrumb"
      className="pf-c-breadcrumb"
      data-ouia-component-id="OUIA-Generated-Breadcrumb-1"
      data-ouia-component-type="PF4/Breadcrumb"
      data-ouia-safe={true}
    >
      <ol
        className="pf-c-breadcrumb__list"
      >
        <BreadcrumbItem
          isActive={false}
          key=".$0"
          showDivider={false}
        >
          <li
            className="pf-c-breadcrumb__item"
          >
            <Link
              to="/clients"
            >
              <LinkAnchor
                href="/clients"
                navigate={[Function]}
              >
                <a
                  href="/clients"
                  onClick={[Function]}
                >
                  <span
                    key="/clients"
                  >
                    Clients
                  </span>
                </a>
              </LinkAnchor>
            </Link>
          </li>
        </BreadcrumbItem>
        <BreadcrumbItem
          isActive={true}
          key=".$1"
          showDivider={true}
        >
          <li
            className="pf-c-breadcrumb__item"
          >
            <span
              className="pf-c-breadcrumb__item-divider"
            >
              <AngleRightIcon
                color="currentColor"
                noVerticalAlign={false}
                size="sm"
              >
                <svg
                  aria-hidden={true}
                  aria-labelledby={null}
                  fill="currentColor"
                  height="1em"
                  role="img"
                  style={
                    Object {
                      "verticalAlign": "-0.125em",
                    }
                  }
                  viewBox="0 0 256 512"
                  width="1em"
                >
                  <path
                    d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
                  />
                </svg>
              </AngleRightIcon>
            </span>
            <span
              key="/clients/1234"
            >
              Client details
            </span>
          </li>
        </BreadcrumbItem>
      </ol>
    </nav>
  </Breadcrumb>
</PageBreadCrumbs>
`;
